qualtran/_infra/registers.py

```diff
@@ -40,7 +43,7 @@ class Side(enum.Flag):
 
 
 @frozen
-class Register:
+class Register(Generic[QDTypeT]):
```

Comment:
I have strong questions about whether this actually needs to be generic.

From a quick inspection, it looks like it's useful in the `registers_test` where you're making registers and then using them and want to access specific properties of the datatype. I fear this has limited utility outside of this niche unless we make the generic-ness pervasive -- i.e. Signature would need to be generic w.r.t. a tuple of QDTypeT or something wild. 

These types are for the quantum program we're reasoning about at python run time. Remember: everything is sortof kicked up a level when we talk about programming concepts. Python class attributes (i.e. runtime values) are quantum program static/template parameters (i.e. concrete values in the quantum "binary"). We want qualtran to do the quantum type checking. We use mypy to do python type checking that we built the library correctly. 

Maybe the python type system can be hijacked to do more static validation of our quantum programs, but I'd strongly suggest removing the generic, using `QDType` (_not_ `QDTypeT`) everywhere, and considering the addition of these generic python types in follow-on work.

Reply:
yeah I didn't find much use for it so far, so it can go.